rust/agents/relayer/src/msg/pending_message.rs

```diff
@@ -297,9 +297,43 @@ impl PendingOperation for PendingMessage {
     fn _next_attempt_after(&self) -> Option<Instant> {
         self.next_attempt_after
     }
+
+    #[cfg(test)]
+    fn set_retries(&mut self, retries: u32) {
+        self.set_retries(retries);
+    }
 }
 
 impl PendingMessage {
+    /// Constructor that tries reading the retry count from the HyperlaneDB in order to recompute the `next_attempt_after`.
+    /// In case of failure, behaves like `Self::new(...)`.
+    pub fn from_persisted_retries(message: HyperlaneMessage, ctx: Arc<MessageContext>) -> Self {
+        fn log_db_read_error(maybe_error: Option<DbError>, message_id: &H256) {
+            let mut log_message = format!("Failed to read retry count from HyperlaneDB for message {}. Defaulting to the regular `PendingMessage` constructor.", message_id);
+            if let Some(error) = maybe_error {
+                log_message.push_str(&format!(" Error: {}", error));
+            }
+            info!(log_message);
```

Comment:
Imo if `maybe_error` is None this should be trace, otherwise something like info or warn makes sense to me. Otherwise I think the default case of a brand new pending message being created will be confusing for ppl